apps/src/blockly/addons/cdoTrashcan.js

```diff
@@ -1,24 +1,236 @@
 import GoogleBlockly from 'blockly/core';
-import {ToolboxType} from '../constants';
 
-export default class CdoTrashcan extends GoogleBlockly.Trashcan {
+export default class CdoTrashcan extends GoogleBlockly.DeleteArea {
+  constructor(workspace) {
+    super();
+    this.workspace = workspace;
+    this.id = 'cdoTrashcan';
+
+    /**
+     * Width of both the trash can and lid images.
+     * @const {number}
+     * @private
+     */
+    this.WIDTH_ = 47;
+    /**
+     * Height of the trashcan image (minus lid).
+     * @const {number}
+     * @private
+     */
+    this.BODY_HEIGHT_ = 44;
+    /**
+     * Height of the lid image.
+     * @const {number}
+     * @private
+     */
+    this.LID_HEIGHT_ = 16;
+    /**
+     * Distance between trashcan and top of toolbox.
+     * @const {number}
+     * @private
+     */
+    this.MARGIN_TOP_ = 40;
+    /**
+     * Location of trashcan in sprite image.
+     * @const {number}
+     * @private
+     */
+    this.SPRITE_LEFT_ = 0;
+    /**
+     * Location of trashcan in sprite image.
+     * @const {number}
+     * @private
+     */
+    this.SPRITE_TOP_ = 32;
+    /**
+     * The length of the lid open/close animation in milliseconds.
+     * @const {number}
+     * @private
+     */
+    this.ANIMATION_LENGTH_ = 80;
+    /**
+     * The number of frames in the animation.
+     * @const {number}
+     * @private
+     */
+    this.ANIMATION_FRAMES_ = 4;
+    /**
+     * The maximum angle the trashcan lid can opens to. At the end of the open
+     * animation the lid will be open to this angle.
+     * @const {number}
+     * @private
+     */
+    this.MAX_LID_ANGLE_ = 45;
+    /**
+     * The minimum (resting) opacity of the trashcan and lid.
+     * @const {number}
+     * @private
+     */
+    this.OPACITY_MIN_ = 0.4;
+    /**
+     * The maximum (hovered) opacity of the trashcan and lid.
+     * @const {number}
+     * @private
+     */
+    this.OPACITY_MAX_ = 0.8;
+    /**
+     * Current open/close state of the lid.
+     * @type {boolean}
+     */
+    this.isLidOpen = false;
```

Comment:
What is the difference between `isLidOpen` and `lidOpen_`?

Reply:
`isLidOpen` is a boolean, while `lidOpen` is a number between 0 and 1 representing the current step in animating the lid open. Not really an angle, but described elsewhere as the current "openness" of the trashcan. This specific label is used to follow conventions established in the [Google Blockly version of this file](https://github.com/google/blockly/blob/2bf842ac65456e61d8c655daec99f8fbc75d2c29/core/trashcan.js#L136).